src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -337,9 +337,9 @@ export default class ReimbursementRequestService {
     });
 
     if (!request) throw new NotFoundException('Reimbursement Request', requestId);
-    if (request.recipientId !== submitter.userId)
+    if (!(await isUserLeadOrHeadOfFinanceTeam(submitter)))
       throw new AccessDeniedException(
-        'You do not have access to delete this reimbursement request, only the creator can delete a reimbursement request'
+        'You do not have access to delete this reimbursement request, only the creator or finance leads can delete a reimbursement request'
```

Comment:
this is actually only checking if the user is finance, not if they're a creator too. add back the if statement you deleted that was there before. also make the message for this say 'only finance leads...'